src/VisualStudio/Core/Def/Implementation/SolutionSize/SolutionSizeTracker.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Concurrent;
+using System.Composition;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.SolutionCrawler;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionSize
+{
+    /// <summary>
+    /// Track approximate solution size.
+    /// </summary>
+    [Export]
+    [ExportIncrementalAnalyzerProvider(WorkspaceKind.Host), Shared]
+    internal class SolutionSizeTracker : IIncrementalAnalyzerProvider
+    {
+        private readonly IncrementalAnalyzer tracker = new IncrementalAnalyzer();
+
+        public long GetSolutionSize(Workspace workspace, SolutionId solutionId)
+        {
+            var vsWorkspace = workspace as VisualStudioWorkspaceImpl;
+            if (vsWorkspace == null)
+            {
+                return -1;
+            }
+
+            return tracker.GetSolutionSize(solutionId);
+        }
+
+        IIncrementalAnalyzer IIncrementalAnalyzerProvider.CreateIncrementalAnalyzer(Workspace workspace)
+        {
+            var vsWorkspace = workspace as VisualStudioWorkspaceImpl;
+            if (vsWorkspace == null)
+            {
+                return null;
+            }
+
+            return tracker;
+        }
+
+        private class IncrementalAnalyzer : IIncrementalAnalyzer
+        {
+            private readonly ConcurrentDictionary<ProjectId, long> _map = new ConcurrentDictionary<ProjectId, long>(concurrencyLevel: 2, capacity: 10);
+
+            private SolutionId _solutionId;
+            private long _size;
+
+            public long GetSolutionSize(SolutionId solutionId)
+            {
+                if (_solutionId != solutionId)
+                {
+                    return -1;
+                }
+
+                return _size;
+            }
+
+            public Task NewSolutionSnapshotAsync(Solution solution, CancellationToken cancellationToken)
+            {
+                if (_solutionId != solution.Id)
+                {
+                    _map.Clear();
+
+                    _solutionId = solution.Id;
+                }
+
+                return SpecializedTasks.EmptyTask;
+            }
+
+            public async Task AnalyzeProjectAsync(Project project, bool semanticsChanged, CancellationToken cancellationToken)
+            {
+                var sum = await GetProjectSizeAsync(project, cancellationToken).ConfigureAwait(false);
+
+                _map.AddOrUpdate(project.Id, sum, (id, existing) => sum);
+
+                _size = _map.Values.Sum(v => v);
```

Comment:
Do you need the ```v => v``` here? There's an overload of .Sum that takes no arguments and appears to do the right thing for IEnumerable<long>